test/tests.transform.js

```diff
@@ -47,3 +47,102 @@ describe("_blockHoist nodes", function() {
     assert.deepEqual(names, ["hoistMe", "doNotHoistMe", "oyez"]);
   });
 });
+
+context("functions", function() {
```

Comment:
@Andarist lol, didn't realize that. 

in our case we're generating api sagas automatically from swagger docs and grouping them by model. we've got at least one saga for every single version / endpoint / method so for some models we have >30 sagas in a single file. most of them aren't used but due to this issue using a single saga from a file pulls in all the others.

in retrospect i probably overstated the value of this to the average user - in normal circumstances i can't imagine having that many sagas sitting around that aren't actually used (although i can imagine situations with code-spitting where you end up bundling an entire module because you want access to a single saga inside of it).